flink-table/flink-sql-parser/src/main/codegen/includes/parserImpls.ftl

```diff
@@ -410,15 +410,54 @@ SqlShowViews SqlShowViews() :
 }
 
 /**
-* Parse a "Show Tables" metadata query command.
+* SHOW TABLES FROM [catalog.] database sql call.
 */
 SqlShowTables SqlShowTables() :
 {
+    SqlIdentifier databaseName;
+    SqlCharStringLiteral likeLiteral = null;
+    String prep = "FROM";
+    boolean notLike = false;
+    SqlParserPos pos;
 }
 {
     <SHOW> <TABLES>
+    [
+        [( <FROM> | <IN> { prep = "IN"; } )]
```

Comment:
Should remove the `[ ]`?  Otherwise, `SHOW TABLES db1` would be valid.

Reply:
done